appcues/src/debug/java/com/appcues/CompositionTestHelper.kt

```diff
@@ -125,6 +127,76 @@ public fun ComposeContainer(context: Context, stepContentJson: List<String>?, tr
+@Composable
+public fun ComposeContainer(
+    context: Context,
+    experienceJson: String,
+    groupIndex: Int,
+    stepIndex: Int,
+    animated: Boolean,
+    imageLoader: ImageLoader,
+) {
+    // set up a Koin scope for testing - for experience/trait mapping, trait registry, etc
+    val scope = AppcuesKoinContext.createAppcuesScope(context, AppcuesConfig("", ""))
+
+    // this JSON is actually a qualify response, and we'll get the first experience out of it
+    val qualifyResponse = MoshiConfiguration.moshi.adapter(QualifyResponse::class.java).fromJson(experienceJson)!!
+    val experienceResponse = qualifyResponse.experiences[0] as ExperienceResponse
+
+    // update the experience to strip out any actions, as they can rely on injected dependencies
+    // that are not available using the test context
+    val updatedExperienceResponse = experienceResponse.copy(
+        steps = experienceResponse.steps.map { stepContainerResponse ->
+            stepContainerResponse.copy(
+                children = stepContainerResponse.children.map { stepResponse ->
+                    stepResponse.copy(
+                        actions = emptyMap()
+                    )
+                },
+                actions = emptyMap()
+            )
+        },
+    )
+
+    // map the experience
+    val experienceMapper: ExperienceMapper = scope.get()
+    val experience = experienceMapper.map(updatedExperienceResponse, ExperienceTrigger.Preview)
+    val container = experience.stepContainers[groupIndex]
+    val metadataSettingTraits = container.steps[stepIndex].metadataSettingTraits
+    val metadata = hashMapOf<String, Any?>().apply { metadataSettingTraits.forEach { putAll(it.produceMetadata()) } }
+
+    AppcuesTheme {
+        CompositionLocalProvider(
+            LocalImageLoader provides imageLoader,
+            LocalLogcues provides Logcues(LoggingLevel.DEBUG),
+            LocalExperienceStepFormStateDelegate provides ExperienceStepFormState(),
+            LocalAppcuesActionDelegate provides FakeAppcuesActionDelegate(),
+            LocalExperienceCompositionState provides ExperienceCompositionState(
+                // disables animations
+                isContentVisible = MutableTransitionState(!animated),
+                isBackdropVisible = MutableTransitionState(!animated)
+            )
+        ) {
+            // render the step container on the desired step
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ) {
+                ComposeContainer(container, stepIndex, metadata)
+            }
+
+            if (animated) {
+                val compositionState = LocalExperienceCompositionState.current
+                LaunchedEffect(Unit) {
+                    compositionState.isContentVisible.targetState = true
+                    compositionState.isBackdropVisible.targetState = true
+                }
+            }
```

Comment:
this allows us to run some tests with full normal animation support, if needed - some experience JSON test do need this (not many)